update_annotations_table.R

```diff
@@ -0,0 +1,77 @@
+################################################
+####  Update the Synapse Annotations table  ####
+################################################
+
+library("jsonlite")
+library("tidyverse")
+library("reticulate")
+library("glue")
+library("httr")
+library("here")
+
+## Log in to synapse
+synapse <- import("synapseclient")
+syn <- synapse$Synapse()
+syn$login()
+
+## Get release version
+release <- GET("https://api.github.com/repos/Sage-Bionetworks/synapseAnnotations/releases")
+body <- content(release)
+release_version <- body[[1]]$tag_name
+
+## JSON files
+files <- list.files(
+  here("synapseAnnotations", "data"),
+  full.names = TRUE
+)
+files <- files[grepl("\\.json$", files)] # no directories
+names(files) <- tools::file_path_sans_ext(basename(files))
+
+## Ensure `value` column gets converted to character, otherwise `unnest()`
+## complains about incompatible types
+value_to_chr <- function(x) {
+  if(is.null(x$value)) {
+    return(NULL)
+  }
+  mutate(x, value = as.character(value))
+}
+
+## Unnest the enumerated values for each key
+unnest_module <- function(x) {
+  x %>%
+    unnest(
+      cols = "enumValues",
+      names_repair = "universal",
+      keep_empty = TRUE
+    ) %>%
+    rename(
+      description = `description...2`,
+      valueDescription = `description...6`,
+      source = `source...7`
+    ) %>%
+    select(-`source...10`)
```

Comment:
This is the source of the key, which currently is unused in the table. Let's comment here - maybe decide if it's useful (Kara notes only ~1 thing has a source for the key).